test/method/testddec.py

```diff
@@ -164,3 +165,31 @@ def test_water_charges(self):
         )
         # Check assigned charges
         assert_allclose(analysis.fragcharges, [-0.757097, 0.378410, 0.378687], atol=0.2)
+
+    def test_chgsum_h2(self):
+        """ Are DDEC6 charges for hydrogen atoms in nonpolar H2 small as expected?
+        """
+
+        h2path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "h2.out")
+        data = ccread(h2path)
+        vol = volume.Volume((-2, -2, -2), (2, 2, 2), (0.2, 0.2, 0.2))
+        analysis = DDEC6(data, vol, os.path.dirname(os.path.realpath(__file__)))
+        analysis.calculate()
+
+        self.assertAlmostEqual(numpy.sum(analysis.fragcharges), 0, delta=1e-2)
+        self.assertAlmostEqual(analysis.fragcharges[0], analysis.fragcharges[1], delta=1e-6)
+
+    def test_chgsum_co(self):
+        """ Are DDEC6 charges for carbon monoxide reported as expected?
+        """
+
+        copath = os.path.join(os.path.dirname(os.path.realpath(__file__)), "co.out")
+        data = ccread(copath)
+        vol = volume.read_from_cube(
+            os.path.join(os.path.dirname(os.path.realpath(__file__)), "co.cube")
+        )
+        analysis = DDEC6(data, vol, os.path.dirname(os.path.realpath(__file__)))
+        analysis.calculate()
+
+        self.assertAlmostEqual(numpy.sum(analysis.fragcharges), 0, delta=1e-2)
+        assert_allclose(analysis.fragcharges, [0.13221636, -0.13903595], atol=1e-3)
```

Comment:
Intuitively, this answer makes more sense than the H2 answer. It's still a little concerning that there is such a deviation from a total of zero (-0.00682). What does chargemol give?

Reply:
This should most likely come from the fact that `self.charge_density.integrate()` returns 14.006876594937234. I've added a note about this. Thanks.

`chargemol` seems to be free from this because they normalize the density to total number of electrons before partitioning them (as in `module_DDEC6_valence_iterator.f08:149`). I'm not sure if this normalization step should be added given that this was not mentioned in the paper?

Reply:
As a follow-up, using finer grid (spacing of 0.05) reduces the difference:

```
>>> ddec.fragcharges
array([ 0.13149808, -0.13052562])
>>> ddec.fragcharges.sum()
0.0009724637203740727
>>> ddec.charge_density.integrate()
13.999027536279629
```